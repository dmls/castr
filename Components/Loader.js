import React from 'react';
import { StyleSheet, View, Modal, ActivityIndicator } from 'react-native';
import { colors } from '../assets/styles/Styles';

const Loader = (props) => {
  const { loading, ...attributes } = props;

  return (
    <Modal
      transparent={true}
      animationType={'none'}
      visible={loading}
    >
      <View style={styles.modalBackground}>
        <View style={styles.activityIndicatorWrapper}>
          <ActivityIndicator
            animating={true}
            color={colors.primary}
            size="large"
            style={styles.activityIndicator}
          />
        </View>
      </View>
    </Modal>
  )
}

export default Loader;

const styles = StyleSheet.create({
  modalBackground: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'center',
    backgroundColor: colors.white,
  },
  activityIndicatorWrapper: {
    height: 100,
    width: 100,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  activityIndicator: {
    alignItems: 'center',
    height: 80,
  },
});
