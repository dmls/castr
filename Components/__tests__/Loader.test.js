import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '../Loader';

describe('Loader', () => {
  test('renders correctly with default settings', () => {
    const component = renderer.create(
      <Loader />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with loading = false', () => {
    const component = renderer.create(
      <Loader loading={false} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with loading = true', () => {
    const component = renderer.create(
      <Loader loading={true} />
    );
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
