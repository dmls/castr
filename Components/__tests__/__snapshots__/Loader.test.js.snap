// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly with default settings 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": undefined,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "height": 100,
          "justifyContent": "space-around",
          "width": 100,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#0d6efd"
        size="large"
        style={
          {
            "alignItems": "center",
            "height": 80,
          }
        }
      />
    </View>
  </View>
</Modal>
`;

exports[`Loader renders correctly with loading = false 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;

exports[`Loader renders correctly with loading = true 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": undefined,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "height": 100,
          "justifyContent": "space-around",
          "width": 100,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#0d6efd"
        size="large"
        style={
          {
            "alignItems": "center",
            "height": 80,
          }
        }
      />
    </View>
  </View>
</Modal>
`;
